src/controller/python/chip/commissioning/commissioning_flow_blocks.py

```diff
@@ -0,0 +1,182 @@
+            (commissioning.ROOT_ENDPOINT_ID, Clusters.BasicInformation.Attributes.VendorID),
+            (commissioning.ROOT_ENDPOINT_ID, Clusters.BasicInformation.Attributes.ProductID)], returnClusterObject=True))[commissioning.ROOT_ENDPOINT_ID][Clusters.BasicInformation]
+
+        self._logger.info("Getting AttestationNonce")
+        attestation_nonce = await self._credential_provider.get_attestation_nonce()
+
+        self._logger.info("Sending AttestationRequest")
+        attestation_elements = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.AttestationRequest(
+            attestationNonce=attestation_nonce
+        ))
+
+        self._logger.info("Getting CertificateChain - DAC")
+        dac = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.CertificateChainRequest(
+            certificateType=1
+        ))
+
+        self._logger.info("Getting CertificateChain - PAI")
+        pai = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.CertificateChainRequest(
+            certificateType=2
+        ))
+
+        self._logger.info("Getting CSR Nonce")
+        csr_nonce = await self._credential_provider.get_csr_nonce()
+
+        self._logger.info("Getting OpCSRRequest")
+        csr = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.CSRRequest(
+            CSRNonce=csr_nonce
+        ))
+
+        self._logger.info("Getting device certificate")
+        commissionee_credentials = await self._credential_provider.get_commissionee_credentials(
+            commissioning.GetCommissioneeCredentialsRequest(
+                dac=dac, pai=pai,
+                attestation_challenge=attestation_nonce,
+                attestation_elements=attestation_elements.attestationElements,
+                attestation_signature=attestation_elements.signature,
+                csr_challenge=csr_nonce,
+                csr_elements=csr.NOCSRElements,
+                csr_signature=csr.attestationSignature,
+                vendor_id=device_info.vendorID,
+                product_id=device_info.productID))
+
+        self._logger.info("Adding Trusted Root Certificate")
+        response = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.AddTrustedRootCertificate(
+            rootCertificate=commissionee_credentials.rcac
+        ))
+
+        self._logger.info("Adding Operational Certificate")
+        response = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.OperationalCredentials.Commands.AddNOC(
+            NOCValue=commissionee_credentials.noc,
+            ICACValue=commissionee_credentials.icac,
+            IPKValue=commissionee_credentials.ipk,
```

Comment:
We should be able to "override" the controller's GroupDataProvider's IPK using the one we get here, right after the AddNOC call, since it will be needed for CASE later.